lib/ralph/irc/connection.ex

```diff
@@ -0,0 +1,91 @@
+defmodule Ralph.IRC.Connection do
+  use GenServer
+
+  def start_link([bot, registry, config]) do
+    GenServer.start_link(__MODULE__, {bot, registry, config})
+  end
+
+  def write(pid, data) do
+    GenServer.call(pid, {:write, data})
+  end
+
+  def do_write(conn, data) do
+    :gen_tcp.send(conn, data) |> IO.inspect()
+  end
+
+  def init({bot, registry, config}) do
+    IO.inspect(config, label: "in init")
```

Comment:
Init should be fast. Defer connections to the network(s) in a `handle_info` timeout block.